src/main/webapp/app/exercises/text/assess/text-assessment.service.ts

```diff
@@ -106,22 +106,24 @@ export class TextAssessmentService {
         } else {
             params = params.set('correction-round', correctionRound.toString());
         }
-        return this.http.get<StudentParticipation>(`${this.resourceUrl}/submission/${submissionId}`, { observe: 'response', params }).pipe(
-            // Wire up Result and Submission
-            tap((response) => {
-                const participation = response.body!;
-                const submission = participation.submissions![0];
-                let result;
-                if (resultId) {
-                    result = getSubmissionResultById(submission, resultId);
-                } else {
-                    result = getSubmissionResultByCorrectionRound(submission, correctionRound)!;
-                }
-                TextAssessmentService.reconnectResultsParticipation(participation, submission, result!);
-                (submission as TextSubmission).atheneTextAssessmentTrackingToken = response.headers.get('x-athene-tracking-authorization') || undefined;
-            }),
-            map((response) => response.body!),
-        );
+        return this.http
+            .get<StudentParticipation>(`${this.resourceUrl}/text-submissions/${submissionId}/for-assessment`, { observe: 'response', params })
+            .pipe<HttpResponse<StudentParticipation>, StudentParticipation>(
+                // Wire up Result and Submission
+                tap((response) => {
```

Comment:
I actually meant `response: HttpResponse<StudentParticipation>` as parameter. 😅 But I suppose this works too?